apps/data/test/lib/opportunities_test.exs

```diff
@@ -21,6 +21,16 @@ defmodule Data.OpportunitiesTest do
     assert %{id: _id} = Opportunities.get(%{url: url})
   end
 
+  test "sort opportunities" do 
+    insert(:opportunity, title: "B Example")
+    insert(:opportunity, title: "C Example")
+    insert(:opportunity, title: "A Example")
+
+    result = Enum.sort(Opportunities.all)
+
+    assert result = Opportunities.all(%{sort_by: :title}) 
```

Comment:
We should try to test the results a bit more explicitly, how about we capture the opportunities returned from `all/1` and retry to match them to the order expect?

Something like this:

```elixir
%{entries: opportunities} = Opportunities.all(%{sort_by: :title})
assert [%{title: "A Example"}, %{title: "B Example"}, %{title: "C Example"}] = opportunities
```

Reply:
I forgot to answer your other question WRT adding other fields:  I don't think that's necessary since the functionality is the same regardless of the field we pass in currently.

That said, we may want to add a new story (which you can implement in another PR!) to allow us sort `difficulty` in a special way not based on the actual name of the difficulty.  If we use the code as-is today, the results would look something like this (Using `Enum.sort/` as an example):

```elixir
iex> Enum.sort(["Beginner", "Intermediate", "Advanced"])
["Advanced", "Beginner", "Intermediate"]
```

What we _probably_ want is to end up with it sorted as Beginner, Intermediate, Advanced.

Make sense?